src/index.js

```diff
@@ -0,0 +1,139 @@
+  tempDecrease: document.getElementById("temp-decrease"),
+  tempVal: parseInt(document.getElementById("temp").textContent),
+  landscapeEl: document.getElementById("landscape"),
+  mainTitle: document.getElementById("city"),
+  inputField: document.getElementById("city-choice"),
+  getTempButton: document.getElementById("get-current-temp"),
+  skyDropdown: document.getElementById("sky-dropdown"),
+  skyEl: document.getElementById("sky"),
+  resetCityButton: document.getElementById("reset-city"),
+}
+
+// Wave2
+const changeTempColor = () => {
+  if (state.tempVal > 80) {
+    state.tempEl.className = 'above80Temp';
+    state.tempConEl.className = 'above80Bk';
+  } else if (state.tempVal < 80 && state.tempVal > 69) {
+    state.tempEl.className = 'temp69To80';
+    state.tempConEl.className = 'temp69To80Bk';
+  } else if (state.tempVal < 70 && state.tempVal > 59) {
+    state.tempEl.className = 'temp59To70';
+    state.tempConEl.className = 'temp59To70Bk';
+  } else if (state.tempVal < 60 && state.tempVal > 49) {
+    state.tempEl.className = 'temp49To60';
+    state.tempConEl.className = 'temp49To60Bk';
+  } else if (state.tempVal < 50) {
+    state.tempEl.className = 'blow50';
+    state.tempConEl.className = 'blow50Bk';
+  };
+}
+
+const changeLandscape = () => {
+  const landscapeImage = document.getElementById("landscape-img");
+  if (state.tempVal >= 80) {
+    landscapeImage.src = 'assets/landscapes/dunes.jpg';
+    landscapeImage.alt = 'Desert dunes landscape';
+  } else if (state.tempVal < 80 && state.tempVal >= 70) {
+    landscapeImage.src = 'assets/landscapes/grass.jpg';
+    landscapeImage.alt = 'Happy cow in lush grassy green hills landscape';
+  } else if (state.tempVal < 70 && state.tempVal >= 33) {
+    landscapeImage.src = 'assets/landscapes/forest.jpg';
+    landscapeImage.alt = 'Foggy dense forest landscape';
+  } else {
+    landscapeImage.src = 'assets/landscapes/tundra.jpg';
+    landscapeImage.alt = 'Icy expansive landscape';
+  }
+}
+
+state.tempIncrease.addEventListener('click', (event) => {
+  event.preventDefault()
+  state.tempVal += 1;
+  state.tempEl.textContent = state.tempVal;
+  changeTempColor();
+  changeLandscape();
+})
+
+state.tempDecrease.addEventListener('click', (event) => {
+  event.preventDefault()
+  state.tempVal -= 1;
+  state.tempEl.textContent = state.tempVal;
+  changeTempColor();
+  changeLandscape();
+})
+
+//Wave 3
+state.inputField.addEventListener('keyup', (event) => {
+  state.mainTitle.textContent = state.inputField.value
+})
+
+//Wave 4
+let lat;
+let lon;
+const getLocation = () => {
+  // console.log(state.mainTitle);
+  const url = `http://127.0.0.1:5000/location`;
+  axios.get(url, {
+    params: {
+      q: state.mainTitle.textContent,
+    }})
+    .then((response) => {
+      console.log(response.data[0].lat, response.data[0].lon)
+      getWeather(response.data[0].lat, response.data[0].lon);
+      }
+    )
+    .catch((error) => {
+      console.log(error);
+    })
+};
```

Comment:
Here's an alternative to promise chaining! You can use the `async/await` keywords to have your function wait until a promise is fulfilled and then you pass the results of that promise to other pieces of your function like assigning your return values as variables or other things like that. It looks something like this:

```javascript
const findCityLocation = async () => {
    let lat;
    let lon;
    const resp = await axios.get('http://127.0.0.1:5000/location', {
        params: {
            q: document.getElementById("city").value
        }
    })
    
    try {
      [lat, lon] = resp.data[0]
      getWeather({ lat: lat, lon: lon});
    } catch (error) {
       console.log(error)
    }
};
```